File: CS50x-Week1-Solution-With-Deep-Explanation-main/mario (feeling less comfortable)/mario.c
#include <cs50.h> // Include the CS50 library for getting input from the user
#include <stdio.h> // Include the standard input/output library for printing output

int main(void)
{
    int size; // Variable to store the height of the pyramid

    // Prompt the user to enter a height between 1 and 8 (inclusive)
    do {
        size = get_int("Height: "); // Get the height from the user
    } while (size < 1 || size > 8); // Repeat the prompt if the height is not within the valid range

    // Loop to construct each row of the pyramid
    for (int i = 0; i < size; i++)
    {
        // Print spaces before the hashes to align the pyramid to the right
        for (int j = 0; j < size - i - 1; j++)
        {
            printf(" "); // Print a space
        }

        // Print hashes for the current row
        for (int k = 0; k <= i; k++)
        {
            printf("#"); // Print a hash
        }

        // Move to the next line after completing the current row
        printf("\n");
    }

    return 0; // Return 0 to indicate successful completion of the program
}

File: CS50x-Week6-Solution-With-Deep-Explanation-main/mario (feeling less comfortable)/mario.py
while True:
    try:
        # Prompt the user to enter the height of the pyramid
        height = int(input("Height: "))
        # Check if the entered height is within the valid range (1 to 8 inclusive)
        if 1 <= height <= 8:
            break  # Exit the loop if the height is valid
        else:
            # Inform the user that the input is invalid if it's not in the range
            print("Invalid input. Please enter a number between 1 and 8.")
    except ValueError:
        # Handle the case where the user input is not a valid integer
        print("Invalid input. Please enter a valid integer.")

# Generate the pyramid
for i in range(1, height + 1):
    # Construct and print each row of the pyramid
    # ' ' * (height - i) calculates the number of leading spaces needed for the current row
    # '#' * i creates the number of hashes needed for the current row
    # Print the row by concatenating spaces and hashes
    print(' ' * (height - i) + '#' * i)
